python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -253,13 +270,15 @@ def _build_failed_partition_subset(
     }
 
     new_failed_partitions = set()
+    in_progress_partitions = set()
     cursor = None
     for partition, (run_id, event_id) in incomplete_materializations.items():
         if run_id in finished_runs:
             status = finished_runs.get(run_id)
             if status == DagsterRunStatus.FAILURE:
                 new_failed_partitions.add(partition)
         else:
+            in_progress_partitions.add(partition)
```

Comment:
this logic is duplicated for building and updating the cache. planning a followup refactor